### PR TITLE
Building for Linux

### PR DESCRIPTION
**Pull request by anetczuk**
Hello,
I made some changes allowing to build the library for Linux machine. Moreover added few fixes of crashing/segfault occurred during release of resources (freeing memory, closing files).

### CODE REVIEW

**Comment by Loobinex** [comment]
Thanks a lot. I'll try it one of these days.

**Comment by Loobinex** [comment]
Sorry, I forgot about it. Tried it now, and I can still build. So that's good, but not enough to merge since I'm no windows user. I asked @dotted to review.

File: libadikted/globals.h [hunk 1]
```diff
@@ -44,7 +44,14 @@
 #elif defined(BUILD_SINGLE)
 # define DLLIMPORT 
 #else /* Not defined BUILD_DLL */
-# define DLLIMPORT __declspec (dllimport)
+# define DLLIMPORT
+//# define DLLIMPORT __declspec (dllimport)
```
Comment (dotted) [inline libadikted/globals.h:48]: Is that intentional?

File: libadikted/Makefile [hunk 2]
```diff
@@ -94,13 +96,13 @@ make-directories:
...
-	$(LINK) $(if $(filter yes, $(DEBUGME)),$(LFLAGS_DEBUG),$(LFLAGS)) $(LIBS) -o $@ $(OBJ) $(RES)
+$(BIN): $(OBJ) #$(RES)
+	$(LINK) $(if $(filter yes, $(DEBUGME)),$(LFLAGS_DEBUG),$(LFLAGS)) $(LIBS) -o $@ $(OBJ) #$(RES)
 
 obj/%.o: %.c
 	$(CC) $(if $(filter yes, $(DEBUGME)),$(CFLAGS_DEBUG),$(CFLAGS)) $< -o $@
 
 $(RES): $(RC)
-	$(WINDRES) -i $< --input-format=rc -o $@ -O coff
+	#$(WINDRES) -i $< --input-format=rc -o $@ -O coff
```
Comment (dotted) [inline libadikted/Makefile:108]: Intentional?

File: libadikted/Makefile [hunk 3]
```diff
@@ -94,13 +96,13 @@ make-directories:
 	$(MKDIR) bin obj
 
 clean:
-	$(RM) $(OBJ) $(BIN) $(BIN:$(LIBEXT)=$(OTHER_LIBEXT)) $(RES)
+	$(RM) $(OBJ) $(BIN) $(BIN:$(LIBEXT)=$(OTHER_LIBEXT)) #$(RES)
 
-$(BIN): $(OBJ) $(RES)
-	$(LINK) $(if $(filter yes, $(DEBUGME)),$(LFLAGS_DEBUG),$(LFLAGS)) $(LIBS) -o $@ $(OBJ) $(RES)
+$(BIN): $(OBJ) #$(RES)
```
Comment (dotted) [inline libadikted/Makefile:101]: Intentional?

File: libadikted/Makefile [hunk 4]
```diff
@@ -94,13 +96,13 @@ make-directories:
 	$(MKDIR) bin obj
 
 clean:
-	$(RM) $(OBJ) $(BIN) $(BIN:$(LIBEXT)=$(OTHER_LIBEXT)) $(RES)
+	$(RM) $(OBJ) $(BIN) $(BIN:$(LIBEXT)=$(OTHER_LIBEXT)) #$(RES)
```
Comment (dotted) [inline libadikted/Makefile:99]: Intentional?

**Review by dotted — CHANGES_REQUESTED** [review]
Sorry for the delay, thank you @Loobinex for the reminder. Looks mostly OK, I'd like for the `windres` stuff to be conditional to when being built to Windows and not just disabled out right.

File: libadikted/bulcommn.c [hunk 5]
```diff
@@ -215,7 +215,7 @@ short write_bmp_fp_24b(FILE *out, int width, int height, const char *data)
         }
     }
     
-    fclose (out);
+    ///fclose (out);            /// callers of function should close the file
```
[the remark this answers is not in the document]
Comment (dotted) [reply libadikted/bulcommn.c:218]: Created an issue for this #26

[the remark this answers is not in the document]
Comment (anetczuk) [reply libadikted/bulcommn.c]: Reverted in new commit.

File: libadikted/globals.h (hunk 1 above)
Comment (anetczuk) [reply libadikted/globals.h:48]: Intentional, if You don't want to build DLL (absence of BUILD_DLL flag) the You don't need "dllimport" spec. Moreover "dllimport" is not supported for ELF (*.so) files (Linux target). Considering this, BUILD_SINGLE flag seems to be redundant/unnecessary.

File: libadikted/Makefile (hunk 2 above)
Comment (anetczuk) [reply libadikted/Makefile]: Intentional, "windres" is not supported for ELF files (Linux target). Updated code to build $(RES) conditionally under Windows.

File: libadikted/Makefile (hunk 3 above)
Comment (anetczuk) [reply libadikted/Makefile]: Reverted (build RES conditionally)

File: libadikted/Makefile (hunk 4 above)
Comment (anetczuk) [reply libadikted/Makefile]: Reverted (build RES conditionally)

**Review by dotted — APPROVED** [review]
LGTM

**Pull request merged**